spec/controllers/products_controller_spec.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ProductsController, type: :controller do
+  let(:category) { create(:category) }
+  let!(:product) { create(:product, category:) }
+  let(:serialized_product) { ProductSerializer.new(product).serializable_hash }
+  let(:error_message) { 'Product not found' }
+
+  let(:service_result) { instance_double('ServiceResult', success?: true, result: product) }
```

Comment:
which instance does this one used from ? the point of using instance double to validate method of that instance for specific class